pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -554,19 +560,25 @@ func newAWSCloud(config io.Reader, awsServices AWSServices) (*AWSCloud, error) {
 	if len(zone) <= 1 {
 		return nil, fmt.Errorf("invalid AWS zone in config file: %s", zone)
 	}
-	regionName := zone[:len(zone)-1]
+
+	regionName := cfg.Global.Region
+	if len(regionName) <= 1 {
+		regionName = zone[:len(zone)-1]
+	}
+
+	endpoint := cfg.Global.Endpoint
 
 	valid := isRegionValid(regionName)
-	if !valid {
-		return nil, fmt.Errorf("not a valid AWS zone (unknown region): %s", zone)
+	if len(cfg.Global.Region) <= 0 && !valid {
```

Comment:
The `<=` is odd.  I think `len(s) == 0` or `s == ""`

But actually I think you only want to make this check if we computed regionName from the zone, so maybe move it into the `cfg.Global.Region == ""` block above?